evaluation/plotting.py

```diff
@@ -42,91 +36,160 @@ def plot_bar_charts(
+            for bar in bars:
+                yval = bar.get_height()
+                ax.text(bar.get_x() + bar.get_width() / 2, yval, round(yval, 1), ha="center", va="bottom", fontsize=8)
 
-        ax.bar(x_data, y_data, width=width, label=name, color=colors)
-        ax.set_title(name, pad=20)
         if y_max_lim is not None and len(y_max_lim) > i and y_max_lim[i] is not None:
-            ax.set_ylim(0, max(y_max_lim[i], max(y_data)))
+            ax.set_ylim(0, y_max_lim[i] * 1.5)
         else:
             ax.set_ylim(0, np.ceil(max(y_data)) * 1.2)
 
+        ax.set_title(titles[i], pad=20)
         ax.set_ylabel(y_labels[i])
+        ax.set_xticks(x_base)
+        ax.set_xticklabels(all_x_data)
+        ax.legend()
 
-        for j, v in enumerate(y_data):
-            ax.text(j, v * 1.02, str(round(v, 2)), ha="center")
-
-    save_figure(output_path)
+    plt.savefig(output_path)
     plt.close(fig)
 
 
-def plot_box_charts_grid(
-    layout: Tuple[int, int], data: List[List[float]], titles: List[str], y_labels: List[str], output_path: Path
+def plot_multiple_box_charts(
+    layout: Tuple[int, int],
+    data: Dict[str, List[List[float]]],
+    titles: List[str],
+    y_labels: List[str],
+    output_path: Path,
 ):
-    """Plot box charts for the provided data in a single figure."""
+    """Plot box charts for the provided data."""
     if layout[0] * layout[1] != len(data):
-        raise ValueError("Number of data points must match the layout")
+        raise ValueError("Number of plots must match the layout")
 
     fig, axs = plt.subplots(layout[0], layout[1], figsize=(layout[1] * 5, layout[0] * 4))
     fig.tight_layout(pad=3.0)
 
-    if axs is not np.ndarray:
+    if not isinstance(axs, np.ndarray):
         axs = np.array([axs])
 
+    positions = range(0, len(data.keys()))
+    offset = 0.2
+
     for i, ax in enumerate(axs.flat):
-        ax.boxplot(data[i])
+        current_positions = []
+        ax = axs.flat[i]
+        for j, (key, values) in enumerate(data.items()):
+            current_positions = [j + offset]
+            ax.boxplot(
+                values[i],
+                positions=current_positions,
+                boxprops=dict(color=f"C{j}"),
+                medianprops=dict(color="red"),
+                whiskerprops=dict(color=f"C{j}"),
+                capprops=dict(color=f"C{j}"),
+            )
+            ax.plot([], [], color=f"C{j}", label=key)
+        ax.legend()
         ax.set_title(titles[i])
         ax.set_ylabel(y_labels[i])
+        ax.set_xticks(positions)
+        ax.set_xticklabels(data.keys())
 
-    save_figure(output_path)
+    plt.savefig(output_path)
     plt.close(fig)
 
 
-def plot_box_chart(
-    data: List[float],
+def plot_single_box_chart(
+    data: Dict[str, List[List[float]]],
     title: str,
     x_labels: List[str],
     y_label: str,
-    output_path: Path,
     y_lim: Tuple[float, float] = None,
+    output_path: Path = Path("boxplot.png"),
```

Comment:
Same above, you can reference the parameter list of this function in the current main branch.